google/cloud/spanner/commit_result.h

```diff
@@ -43,6 +47,26 @@ struct CommitResult {
   absl::optional<CommitStats> commit_stats;
 };
 
+/**
+ * The result of committing a Transaction containing a batch of mutation
+ * groups.  See the batched form of `Client::CommitAtLeastOnce()`.
+ */
+struct BatchedCommitResult {
+  /// The mutation groups applied in this batch. Each value is an index into
+  /// the `std::vector<Mutations>` passed to `Client::CommitAtLeastOnce()`.
+  std::vector<std::size_t> indexes;
+
+  /// If OK, the Cloud Spanner timestamp at which the transaction committed,
+  /// and otherwise the reason why the commit failed.
+  StatusOr<Timestamp> commit_timestamp;
```

Comment:
Speaking to myself: I wonder if we really need a streaming API in the library.  I know the proto is streaming, but maybe we can simply collect all the results and return all of them.  Hmmm... there is the problem of blocking until they all arrive. Sold.